### PR TITLE
Simplified code for starting_url

### PR DESCRIPTION
**Pull request by naslundx**
Makes Mypy less confused in #50

### CODE REVIEW

File: webtraversallibrary/workflow.py [hunk 1]
```diff
@@ -109,12 +109,13 @@ def __init__(
...
-        # Setup initial stuff
-        self._starting_url = url
-        if isinstance(self._starting_url, str):
-            self._starting_url = {Workflow.SINGLE_TAB: self._starting_url}
-        if isinstance(list(self._starting_url.values())[0], str):
-            self._starting_url = {Workflow.SINGLE_TAB: self._starting_url}  # type: ignore
+        # Setup starting points
+        if isinstance(url, str):
+            url = {Workflow.SINGLE_TAB: url}
+        if isinstance(list(url.values())[0], str):
```
Comment (vonNiklasson) [inline webtraversallibrary/workflow.py:115]: Just checking I'm with what happens here, doesn't this only pass if the line on line 114 is evaluated **OR** if the type of `URL` is either `Dict[str, str]` or `Dict[str, Dict[str, str]]`? Would it be reasonable to use an `elif` on line 115 or is there something else going on?

Comment (vonNiklasson) [reply webtraversallibrary/workflow.py:115]: Ping @naslundx in case my grave digging gets unnoticed.

Comment (naslundx) [reply webtraversallibrary/workflow.py:115]: Thank you for reviewing! :) I understand your confusion, but it's not meant to be `elif`. The input may be str, dict[str, str], or dict[str, dict[str, str]]. The first if-statement goes from type 1 to 2, and this one from 2 to 3.

Comment (vonNiklasson) [reply webtraversallibrary/workflow.py:115]: That makes sense, and is indeed a neat parameter assertion. I've opened #59 to ensure no such confusion occurs in the future if it seems like a nice addition to the collection.

**Pull request merged**